### PR TITLE
Add Seq2Seq Chinese model support

### PR DESCRIPTION
**Pull request by zhtmike**
1. Add Chinese model configure file
2. Support pretrained weight with backbone only. Use `remove_prefix=True` to load pretrained model from MindOCR
3. Support relative path of dictionary in modelArt 
4. Add Documentation of training custom dataset
5. Merge the documentation of Chinese dataset from #280 

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        |- word_001.png
+        |- word_002.jpg
+        |- word_003.jpg
+        | ...
+```
+
+## 字典准备
+
+为训练中、英文等不同语种的识别网络，用户需配置对应的字典。只有存在于字典中的字符会被模型正确预测。MindOCR现提供中、英两种字典，其中
+- `英文字典`：包括大小写英文、数字和标点符号。存放于`mindocr/utils/dict/en_dict.txt`
```
Comment (SamitHuang) [inline docs/cn/tutorials/training_recognition_custom_dataset_CN.md:55]: 目前已训练的crnn英文识别模型(trained on MJ+ST) 所用的字典是默认的小写字母+数字，未用到这个字典？
若在自定义数据集上改用这个英文字典，其他超参如num_classes, use_space_char 也要说明如何调整。

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+- `中文字典`：包括常用中文字符、大小写英文、数字和标点符号。存放于`mindocr/utils/dict/ch_dict.txt`
+
+目前MindOCR暂未提供自定义字典配置。该功能将在新版本中推出。
```
Comment (SamitHuang) [inline docs/cn/tutorials/training_recognition_custom_dataset_CN.md:58]: 可否通过指定character_dict_path，并修改num_classes，来使用自定义字典 并 微调训练？

**Review by SamitHuang — COMMENTED** [review]
Looks good overall! Some improvements are needed before merge as mentioned in comments.

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md (hunk 1 above)
Comment (zhtmike) [reply docs/cn/tutorials/training_recognition_custom_dataset_CN.md]: 已修改为新的字典，num_classes, use_space_char 说明已更新。

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md (hunk 2 above)
Comment (zhtmike) [reply docs/cn/tutorials/training_recognition_custom_dataset_CN.md]: 已提供修改字典字符教学。

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    label_file: gt_validation.txt                                     # 训练数据集标签摆放位置，将与`dataset_root`拼接形成完整路径
+  ...
+```
+
+并修改对应的字典位置，指向英文字典路径
+
+```yaml
+...
+common:
+  character_dict_path: &character_dict_path mindocr/utils/dict/en_dict.txt
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply docs/cn/tutorials/training_recognition_custom_dataset_CN.md:92]: 这个初始配置为`crnn_resnet34_ch.yaml`，在PR #280 , `num_claases`应该正确

File: mindocr/models/backbones/builder.py [hunk 4]
```diff
@@ -53,7 +52,12 @@ def build_backbone(name, **kwargs):
     if 'pretrained' in kwargs:
         pretrained = kwargs['pretrained']
         if not isinstance(pretrained, bool):
-            load_model(backbone, pretrained)
+            if remove_prefix:
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply mindocr/models/backbones/builder.py:55]: 应该不行，auto_mapping会找最相近的matches，但如果想在backbone load minocr 模型的话需要把前缀backbone.去掉，差异会有9个字符

File: mindocr/models/backbones/rec_resnet.py [hunk 5]
```diff
@@ -156,15 +156,12 @@ def construct(self, x):
...
-
 
 @register_backbone
 def rec_resnet34(pretrained: bool = True, **kwargs):
     model = RecResNet(in_channels=3, layers=34, **kwargs)
 
-    # load pretrained weights
-    if pretrained:
+    if pretrained is True:
         raise NotImplementedError
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply mindocr/models/backbones/rec_resnet.py]: 已更新

**Comment by zhtmike** [comment]
有两个问题
1. 英文字典和中文字典不统一。英文字典含有空格，中文字典不包含空格，什么时候用use_space_char会有混淆。
2. num_classes应该自动产生，否则要根据字典字符数量、use_space_char 还有模型类别决定，用户用会比较困难去理解

**Comment by SamitHuang** [comment]
1. 字典文件里，不应该包含空格。统一用 use_space_char来指定添加空格支持。
2. num_classes 确实可计算出来（如设为空 / Null)，自动计算。

**Comment by zhtmike** [comment]
> 1. 字典文件里，不应该包含空格。统一用 use_space_char来指定添加空格支持。
> 2. num_classes 确实可计算出来（如设为空 / Null)，自动计算。

1. 已在英文字典去除空格。2 涉及改动较大，建议另开一个PR

**Review by SamitHuang — APPROVED** [review]

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+### 训练集准备
+请将所有训练图片置入同一文件夹，并在上层路径指定一个txt文件用来标注所有训练图片名和对应标签。txt文件例子如下
+
+```
+# 文件名	# 对应标签
+word_421.png	菜肴
+word_1657.png	你好
+word_1814.png	cathay
+```
+*注意*：请将图片名和标签以 \tag 作为分隔，避免使用空格或其他分隔符。
```
Comment (HaoyangLee) [inline docs/cn/tutorials/training_recognition_custom_dataset_CN.md:25]: 此处应为`\tab`或`\t`

File: docs/en/tutorials/training_recognition_custom_dataset.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+### Preparing Training Data
+Please place all training images in a single folder, and specify a txt file at a higher directory to label all training image names and corresponding labels. An example of the txt file is as follows:
+
+```
+# File Name	# Corresponding label
+word_421.png	菜肴
+word_1657.png	你好
+word_1814.png	cathay
+```
+*Note*: Please separate image names and labels using \tag, and avoid using spaces or other delimiters.
```
Comment (HaoyangLee) [inline docs/en/tutorials/training_recognition_custom_dataset.md:25]: 此处应为`\tab`或`\t`

File: configs/rec/rare/README_CN.md
Comment (HaoyangLee) [inline configs/rec/rare/README_CN.md:1]: line19参考文献的引用应修改为："#references" -> "#参考文献"

File: configs/rec/rare/rare_resnet34_ch.yaml [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  use_space_char: &use_space_char False
+  batch_size: &batch_size 512
+
+model:
+  type: rec
+  transform: null
+  backbone:
+    name: rec_resnet34
+    pretrained: https://download.mindspore.cn/toolkits/mindocr/rare/rare_resnet34-309dc63e.ckpt
+    remove_prefix: True
```
Comment (HaoyangLee) [inline configs/rec/rare/rare_resnet34_ch.yaml:24]: 此处建议添加`remove_prefix`参数的注释，不然用户有点难理解

File: docs/cn/datasets/chinese_text_recognition_CN.md
Comment (HaoyangLee) [inline docs/cn/datasets/chinese_text_recognition_CN.md:1]: `docs/`下面`tutorials/`和`datasets/`中更新的文档是否需要在外部文档（如主页README，或RARE模型的REAMDE）给一个链接引过来？否则用户找不到这几个文档

File: docs/cn/tutorials/training_recognition_custom_dataset_CN.md (hunk 6 above)
Comment (zhtmike) [reply docs/cn/tutorials/training_recognition_custom_dataset_CN.md]: 已修改。

File: docs/en/tutorials/training_recognition_custom_dataset.md (hunk 7 above)
Comment (zhtmike) [reply docs/en/tutorials/training_recognition_custom_dataset.md]: 已修改。

File: configs/rec/rare/README_CN.md
Comment (zhtmike) [reply configs/rec/rare/README_CN.md:1]: 已修改。

File: configs/rec/rare/rare_resnet34_ch.yaml (hunk 8 above)
Comment (zhtmike) [reply configs/rec/rare/rare_resnet34_ch.yaml]: 已加注释

File: docs/cn/datasets/chinese_text_recognition_CN.md
Comment (zhtmike) [reply docs/cn/datasets/chinese_text_recognition_CN.md:1]: RARE模型的README已提供连接，等下CRNN 的PR也会指过去。

**Review by HaoyangLee — APPROVED** [review]
